src/client/googlePubSub.ts

```diff
@@ -525,4 +529,23 @@ export default class GooglePubSubAdapter implements PubSubClientV2 {
     );
     return subscriptions.flat();
   }
+
+  public async getAllSubscriptionsOpenState(): Promise<IsOpenTuple[]> {
+    const subscriptions = await Bluebird.map(
+      Object.keys(this.projects),
+      async (project) => {
+        const [subscriptions] = await this.projects[
+          project
+        ].client.getSubscriptions();
```

Comment:
We can use the cached subscriptions map, that way whole thing would sync (no need for Bluebird) and quicker
```suggestion
        const subscriptions = Array.from(
          this.projects[project].subscriptions.values(),
        );
```